notebooks/03-query/15-pivot-unpivot.ipynb

```diff
@@ -0,0 +1,372 @@
+    ")\n",
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  \"isRobot\",\n",
+    "  \"channel\",\n",
+    "  \"flags\",\n",
+    "  \"isUnpatrolled\",\n",
+    "  \"page\",\n",
+    "  \"diffUrl\",\n",
+    "  \"added\",\n",
+    "  \"comment\",\n",
+    "  \"commentLength\",\n",
+    "  \"isNew\",\n",
+    "  \"isMinor\",\n",
+    "  \"delta\",\n",
+    "  \"isAnonymous\",\n",
+    "  \"user\",\n",
+    "  \"deltaBucket\",\n",
+    "  \"deleted\",\n",
+    "  \"namespace\",\n",
+    "  \"cityName\",\n",
+    "  \"countryName\",\n",
+    "  \"regionIsoCode\",\n",
+    "  \"metroCode\",\n",
+    "  \"countryIsoCode\",\n",
+    "  \"regionName\"\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "display_client.run_task(sql)\n",
+    "sql_client.wait_until_ready('example-pivot-unpivot')\n",
+    "display_client.table('example-pivot-unpivot')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "841a8fc2-2efa-4327-b35c-e64a8fa380d3",
+   "metadata": {},
+   "source": [
+    "## Robotic updates to wikipedia by channel\n",
+    "If you want to determine which channels receive the most robotic changes you could use a regular aggregate query like this one:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "79bac420-4ee7-4f77-abdf-be651e5752ef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  \"channel\",\n",
+    "  \"isRobot\",\n",
+    "  SUM(added) total_added,\n",
+    "  SUM(deleted) total_deleted\n",
+    "FROM \"example-pivot-unpivot\" \n",
+    "GROUP BY 1, 2\n",
+    "'''\n",
+    "display_client.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a8642f39-ad04-45b3-9120-b622a0f7c8a7",
+   "metadata": {},
+   "source": [
+    "## PIVOT example\n",
+    "The prior result is a bit long and makes it hard to figure out which channels have the most updates while still seeing the distinction between additions and deletions to the wikipedia pages.\n",
+    "One way to reduce the result while still keeping all the detail is to pivot the result such that rows become columns for each value of a particular column, the pivot column.\n",
+    "\n",
+    "PIVOT is a SQL operator that is used after the FROM clause.\n",
+    "\n",
+    "```\n",
+    "    FROM \"example-pivot-unpivot\" \n",
+    "    \n",
+    "    PIVOT ( SUM(added) as added, \n",
+    "            SUM(deleted) as deleted \n",
+    "            \n",
+    "            FOR \"isRobot\" IN ('true' as robot, 'false' as human)\n",
+    "          )\n",
+    "```\n",
+    "- It lists a set of aggregations to calculate: `SUM(added) as added, SUM(deleted) as deleted`\n",
+    "- The FOR clause indicated the pivot column: `\"isRobot\"`\n",
+    "- The IN clause lists the pivot values of the column and what to call them: `IN ('true' as robot, 'false' as human)`\n",
```

Comment:
```suggestion
    "- The IN clause lists the pivot values of the column and what to call them: `IN ('true' as robot, 'false' as human)`\n",
```
Can we say "column names" instead of "what to call them"?

Reply:
its not the resulting column name, it is the value name which is then used to build a column name.